src/bot.ts

```diff
@@ -423,28 +440,69 @@ export class DiscordBot {
     });
   }
 
-  private async SendMatrixMessage(matrixMsg: MessageProcessorMatrixResult, chan: Discord.Channel,
-                                  guild: Discord.Guild, author: Discord.User,
+  private async StoreMatrixEvent(res: any, room: string, chan: Discord.GuildChannel, msgID: string) {
+      const dbEvt = new DbEvent();
+      dbEvt.MatrixId = res.event_id + ";" + room;
+      dbEvt.DiscordId = msgID;
+      dbEvt.ChannelId = chan.id;
+      dbEvt.GuildId = chan.guild.id;
+      return this.store.Insert(dbEvt).catch((err) => {
+          log.warn("Failed to insert sent event into the database", err);
+      });
+  }
+
+  private async SendMatrixMessage(matrixMsg: MessageProcessorMatrixResult,
+                                  chan: Discord.GuildChannel,
+                                  author: Discord.User|Discord.GuildMember,
                                   msgID: string): Promise<boolean> {
     const rooms = await this.GetRoomIdsFromChannel(chan);
     const intent = this.GetIntentFromDiscordMember(author);
-
-    rooms.forEach((room) => {
-      intent.sendMessage(room, {
-        body: matrixMsg.body,
-        msgtype: "m.text",
-        formatted_body: matrixMsg.formattedBody,
-        format: "org.matrix.custom.html",
-      }).then((res) => {
-        const evt = new DbEvent();
-        evt.MatrixId = res.event_id + ";" + room;
-        evt.DiscordId = msgID;
-        evt.ChannelId = chan.id;
-        evt.GuildId = guild.id;
-        return this.store.Insert(evt);
-      });
-    });
-
+    const textEvt = {
+      body: matrixMsg.body,
+      msgtype: "m.text",
+      formatted_body: matrixMsg.formattedBody,
+      format: "org.matrix.custom.html",
+    };
+
+    while (rooms.length > 0) {
+        const attachments = Array.from(matrixMsg.attachmentEvents);
```

Comment:
So this PR does both discord DMs and discord attachments?

Reply:
It refactored SendMatrixMessage a smidge